internal/pkg/manifest/svc.go

```diff
@@ -513,19 +513,28 @@ func (cfg NetworkLoadBalancerListener) exposedPorts(exposedPorts []ExposedPort,
 	if cfg.TargetPort != nil {
 		targetPort = uint16(aws.IntValue(cfg.TargetPort))
 	}
+	targetProtocol := strings.ToLower(TCP)
+	if nlbProtocol != nil {
+		protocol := strings.ToLower(aws.StringValue(nlbProtocol))
+		// Expose TCP port for TLS listeners.
+		if protocol != strings.ToLower(TLS) {
+			targetProtocol = protocol
+		}
+	}
 	for _, exposedPort := range exposedPorts {
-		if targetPort == exposedPort.Port {
+		if targetPort == exposedPort.Port && targetProtocol == exposedPort.Protocol {
```

Comment:
nit:
```suggestion  
                 // NOTE: ECS doesn't support exposing the same port twice.
                 // This is a feature request to support multiple protocols on the same port: https://github.com/aws/containers-roadmap/issues/850.
		if targetPort == exposedPort.Port  {
```
ECS doesn't let you expose the same port twice, even if they are different protocols. Although as a client side tool we don't have to do the same validations as the service, I think I'd prefer erroring out early here.

NLB can listen on a port for `TPC_UDP` traffic - judging by this design pattern, it is possible that when ECS does support this feature, they will support an "additional protocol" called `TCP_UDP`, instead of letting people expose the same port twice. This is my own personal intuition though. Here is a related feature request to ECS: https://github.com/aws/containers-roadmap/issues/850. 

Anyway, based on this observation, it's likely that we won't need to change this if statement even after they'e supported the feature. So maintenance-wise, this probably won't cause burden in the future anyway.

Reply:
Sure, that makes sense. I'll add that note and revert the change there.

Wouldn't this just silently fail? No `err` is being returned, but I'm not sure if consumers of this method would interpret a `nil` `[]ExposedPort` as a failure.

Reply:
My bad!! I somehow was thinking this code was written in `manifest/validate.go`, and was trying to point out that the "uniqueness" of a port is determined by the port only, not port + protocol.

Now that I've had my coffee, I realize this comment doesn't make sense for the code here. This function is called  after we've done all the manifest validation. At this point, it should be assumed that the manifest is totally valid - that is, no port is exposed twice.

Therefore, I think we can keep it as you have (port + protocol). Since the validation already happened, there is effectively no difference between `if targetPort == exposedPort.Port` and `if targetPort == exposedPort.Port && targetProtocol == exposedPort.Protocol`; however, the latter is clearly and makes more sense 😂 

The validation code already checks that "no port is exposed twice" - it doesn't take protocol into consideration. So we are good there as well.

Sorry for the confusion!!

Reply:
hey there @tjhorner ! Do you want me to take over the code and address the nit, so that this PR can be merged before our next release?

Reply:
Sure, that's fine. Thanks!